### PR TITLE
Update FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql

### PR DESCRIPTION
**Pull request by sailendrasingh1**
Amendment 23.1 SP 1
Date: 08/29/2023

### CODE REVIEW

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 1]
```diff
@@ -242,13 +245,24 @@ IF OBJECT_ID('tempdb..#GeneratedText') IS NOT NULL
...
+	)  
 
 INSERT INTO #GeneratedText 
-SELECT  D.DEATH_REC_ID,D.INTERNAL_CASE_NUMBER,D.SFN_NUM,NULL
-,CONVERT(VARCHAR,YEAR(D.DOD)) + CONVERT(VARCHAR,D.SFN_NUM) + '0'+ CONVERT(CHAR(1),D.VRV_REC_REPLACE_NBR)+ ISNULL(REPLACE(CONVERT(VARCHAR,DATEADD(DAY,+2,CONVERT(DATETIME,D.DATE_OF_AMENDMENT)),101),'/',''),'99999999')+  '1'+ --2
+SELECT  D.DEATH_REC_ID,D.INTERNAL_CASE_NUMBER,D.SFN_NUM,NULL,0 AS RECORD_SORTING
+,CONVERT(VARCHAR,YEAR(D.DOD_4_FD)) + CONVERT(VARCHAR,D.SFN_NUM) + '0'+ CONVERT(CHAR(1),D.VRV_REC_REPLACE_NBR)+ ISNULL(REPLACE(CONVERT(VARCHAR,DATEADD(DAY,+2,CONVERT(DATETIME,D.DATE_OF_AMENDMENT)),101),'/',''),'99999999')+  '1'+ --2
 '|' + ISNULL(CONVERT(VARCHAR,DATEADD(DAY,+2,CONVERT(DATETIME,D.DATE_OF_AMENDMENT)),112),'99999999') + '6666' +  --3
 '|' + 'Electronic' + --4
 '|' + 'BASIS:' + ISNULL(A.MOD_BASIS,'') + ',' + 'REGISTRATON NUMBER:'+ISNULL(D.RECORD_REGIS_NUM,'') + ',' + 'REASON:'+ ISNULL(D.AMENDMENT_REASON ,'BLANK')+ ',' + 'USER ID:'+ ISNULL(U.USERNAME,'') + ','
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:268]: 'REASON:'+ ISNULL(D.AMENDMENT_REASON ,'BLANK') why?
I think it should be  'REASON:'+ ISNULL(D.AMENDMENT_REASON ,'BLANK')

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 2]
```diff
@@ -1570,38 +639,106 @@ AND D.SFN_NUM NOT IN ('000001',
...
+					
+					WHEN c.Field = 'REGISTRATION_DEATH_DEATH_BIRTH_EVENT_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE '99/99/9999' THEN '999999996666' 					
+					WHEN c.Field = 'REGISTRATION_DEATH_DEATH_BIRTH_EVENT_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 1 THEN CONVERT (VARCHAR(8),DATEADD(DAY,+2,CONVERT(DATETIME, AD.ORIGINAL_VALUE)),112) + '6666' 
+					WHEN c.Field = 'REGISTRATION_DEATH_DEATH_BIRTH_EVENT_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 THEN REPLACE ((RIGHT(D.DOB,4) + LEFT(AD.ORIGINAL_VALUE,6)),'/','') + '6666' 
+
+					--FOR VET1_DATE_ENTERED 
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE '99/99/9999' THEN '999999996666' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE IS NULL THEN ''
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 AND REPLACE(AD.ORIGINAL_VALUE,'9','') IN ('/', '') THEN '99999999' + '6666'   -- dont need toi part only keep 6666
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:691]: WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 AND REPLACE(AD.ORIGINAL_VALUE,'9','') IN ('/', '') THEN '99999999' + '6666'
mac do we have this logic in generic load code?

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 3]
```diff
@@ -1570,38 +639,106 @@ AND D.SFN_NUM NOT IN ('000001',
...
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 THEN  REPLACE ((RIGHT(AD.ORIGINAL_VALUE,4) + LEFT(AD.ORIGINAL_VALUE,6)),'/','') + '6666'
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE) = 1 THEN ISNULL(CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME, AD.ORIGINAL_VALUE)),112),'')  + '6666'
+					
+					--FOR VET2_DATE_ENTERED 
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_2_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE '99/99/9999' THEN '999999996666' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_2_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE IS NULL THEN ''
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_2_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 AND REPLACE(AD.ORIGINAL_VALUE,'9','') IN ('/', '') THEN '99999999' + '6666'   -- dont need toi part only keep 6666
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:698]: Same as above

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 4]
```diff
@@ -1570,38 +639,106 @@ AND D.SFN_NUM NOT IN ('000001',
...
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_2_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 THEN  REPLACE ((RIGHT(AD.ORIGINAL_VALUE,4) + LEFT(AD.ORIGINAL_VALUE,6)),'/','') + '6666'
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_2_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE) = 1 THEN ISNULL(CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME, AD.ORIGINAL_VALUE)),112),'')  + '6666'
+					
+					--FOR VET3_DATE_ENTERED 
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_3_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE '99/99/9999' THEN '999999996666' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_3_MILITARY_START_DATE_LONG' AND AD.ORIGINAL_VALUE IS NULL THEN ''
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_3_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 AND REPLACE(AD.ORIGINAL_VALUE,'9','') IN ('/', '') THEN '99999999' + '6666'   -- dont need toi part only keep 6666
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:705]: SSA

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 5]
```diff
@@ -1570,38 +639,106 @@ AND D.SFN_NUM NOT IN ('000001',
...
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_3_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE)= 0 THEN  REPLACE ((RIGHT(AD.ORIGINAL_VALUE,4) + LEFT(AD.ORIGINAL_VALUE,6)),'/','') + '6666'
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_3_MILITARY_START_DATE_LONG' AND ISDATE(AD.ORIGINAL_VALUE) = 1 THEN ISNULL(CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME, AD.ORIGINAL_VALUE)),112),'')  + '6666'
+					
+
+					--FOR VET1_DATE_DISCHARGED
+					
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_DISCHARGE_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE '99/99/9999' THEN '999999996666' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_DISCHARGE_DATE_LONG' AND AD.ORIGINAL_VALUE IS NULL THEN ''
+					WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_DISCHARGE_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE 'N%A' THEN '666666666666'
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:714]: WHEN c.Field = 'TASK_DECEDENT_MILITARY_SERVICE_1_MILITARY_DISCHARGE_DATE_LONG' AND AD.ORIGINAL_VALUE LIKE 'N%A' THEN '666666666666'
Does this exist in Generic load code?

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 6]
```diff
@@ -3126,15 +1728,42 @@ INSERT INTO #GeneratedText
 							ELSE REPLACE(ISNULL(dt.TOD,'6666'),':','') END) AS VARCHAR(10))
 						END))
 					  WHEN c.Field = 'REGISTRATION_DEATH_INJURY_INJURY_DATE_LONG' 
-					  THEN (CASE WHEN dt.DOI = '99/99/9999' THEN ''
+					  THEN (CASE WHEN dt.DOI = '99/99/9999' THEN '999999996666'
 								 WHEN dt.DOI IS NULL THEN ''
+								 WHEN ISDATE(dt.DOI)= 0 AND REPLACE(dt.DOI,'9','') IN ('/', '') THEN '99999999' + CONVERT( VARCHAR,REPLACE(ISNULL(dt.TOI,'6666'),':',''))
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1733]: WHEN ISDATE(dt.DOI)= 0 AND REPLACE(dt.DOI,'9','') IN ('/', '') THEN '99999999' + CONVERT( VARCHAR,REPLACE(ISNULL(dt.TOI,'6666'),':',''))
Is this logic exist in the generic load code?

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 7]
```diff
@@ -3126,15 +1728,42 @@ INSERT INTO #GeneratedText
...
+						(CASE WHEN dt.PRO_DATE IS NULL THEN '' ELSE CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT(DATETIME , dt.PRO_DATE)),112 )
+							+(CASE WHEN dt.PRO_TIME_IN = 'D' AND dt.PRO_TIME = '12:00' THEN '0000'ELSE 
+								CAST((CASE WHEN dt.PRO_TIME_IN = 'P'  AND dt.PRO_TIME NOT LIKE '12:%' THEN CAST(CAST(LEFT(REPLACE(dt.PRO_TIME,':',''),2) AS INT)+12 AS VARCHAR(100)) +CAST(RIGHT(REPLACE(dt.PRO_TIME,':',''),2) AS VARCHAR(100))
+								ELSE REPLACE(ISNULL(dt.PRO_TIME,'6666'),':','') 
+								END) AS VARCHAR(10))
+							END)
+						END	)
+							WHEN C.Field = 'TASK_INFORMANT_ADDRESS_STREET_NAME' THEN
+					(CASE WHEN  dt.INFO_ADDR1 IS NOT NULL AND dt.INFO_PO_BOX IS NOT NULL
+							THEN ISNULL(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(REPLACE(dt.INFO_ADDR1,'A','C'),'E','X'),'I','Q'),'O','Z'),'U','T'),'B','Q'),'')
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1747]: Why do you scramble ADD1?

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 8]
```diff
@@ -3167,1068 +1823,510 @@ INSERT INTO #GeneratedText
...
+		WHEN DOC.SUPPORT1_DATE IS NOT NULL AND DOC.SUPPORT1_TYPE_CODE IS NULL THEN '' 
+		WHEN ISDATE(DOC.SUPPORT1_DATE)= 0 THEN ''
+		WHEN ISDATE(DOC.SUPPORT1_DATE)= 1  THEN ISNULL(CONVERT(VARCHAR ,DATEADD(DAY,+2,CONVERT(DATETIME,DOC.SUPPORT1_DATE )),112) + '6666','999999996666') --WHAT TO DO WITH ISNULL
+		ELSE ''
+	  END + --5
+'|' + CASE
+		WHEN DOC.SUPPORT1_DESCRIPTION IS NOT NULL AND DOC.SUPPORT1_TYPE_CODE IS NULL THEN ''
+		
+		 ELSE ISNULL (REPLACE(REPLACE(DOC.SUPPORT1_DESCRIPTION, Char(13),''),char(10) ,''),'') 
+			+ ' ' + CASE WHEN ISNULL(DOC.SUPPORT1_DESCRIPTION,'') <> '' AND ISDATE(DOC.SUPPORT1_DATE)= 0 AND DOC.SUPPORT1_DATE IS NOT NULL THEN ': Support Document Date (VIP) -' + DOC.SUPPORT1_DATE
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1961]: put comment here.

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql [hunk 9]
```diff
@@ -3167,1068 +1823,510 @@ INSERT INTO #GeneratedText
...
+SELECT * , ROW_NUMBER() over (partition by Death_Rec_Id, DAVEFILEOUT order by AUDIT_RECORD_DETAILS_ID) as row_id 
+from #GeneratedText_DuplicateDate) s
+where row_id =1
 
+select * from [DAVE_MIGRATION].[TRAN_AMEND_GENERATEDTEXT] ORDER BY 1,RECORD_SORTING
 
 
+select * from [DAVE_MIGRATION].[TRAN_AMEND_GENERATEDTEXT]
+--WHERE SFN_NUM = '030965'
+ORDER BY 1,RECORD_SORTING
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:2330]: remove this one. it was already added above
select * from [DAVE_MIGRATION].[TRAN_AMEND_GENERATEDTEXT]
--WHERE SFN_NUM = '030965'
ORDER BY 1,RECORD_SORTING

**Review by almabekwoldesenbet — CHANGES_REQUESTED** [review]
Logically this request does not require. But it's nice to organize. The following fields should be moved in the first part. They have the same scramble logic
TASK_DECEDENT_ADDRESS_STREET_NAME,TASK_PLACE_OF_DEATH_ADDRESS_STREET_NAME

REMOVE TASK_DECEDENT_ADDRESS_STREET_NUMBER' FROM THE ABOVE SECTION IT WILL BE HANDLED IN THE ELSE 

Nice to do: to cleanup code
can you track the list of SFN_Num in another file.  The code looks like huge.


Logically I donot see any issue but make sure the logic is the same as the generic load code

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 1 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:268]: Fixed

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 2 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:691]: Yes. The logic is written like 
WHEN ISDATE(VET1_DATE_ENTERED)= 0 AND REPLACE(VET1_DATE_ENTERED,'9','') IN ('/', '') THEN '99999999' + '6666'

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 3 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:698]: Yes. The logic is written like
 WHEN ISDATE(VET2_DATE_ENTERED)= 0 AND REPLACE(VET2_DATE_ENTERED,'9','') IN ('/', '') THEN '99999999' + '6666'

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 4 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:705]: Yes. The logic is written like
WHEN ISDATE(VET3_DATE_ENTERED)= 0 AND REPLACE(VET3_DATE_ENTERED,'9','') IN ('/', '') THEN '99999999' + '6666'

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 5 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:714]: Yes. The logic is written like
 WHEN VET1_DATE_DISCHARGED LIKE 'N%A' THEN '666666666666'

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 6 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1733]: Yes, the logic is written like 
WHEN ISDATE(DOI)= 0 AND REPLACE(DOI,'9','') IN ('/', '') THEN '99999999' + CONVERT( VARCHAR,REPLACE(ISNULL(TOI,'6666'),':',''))

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 7 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1747]: previously scrambled, but can be removed

Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1747]: Informant address is scrambled in generic script as well hence this has to be scrambled too.

File: RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql (hunk 8 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_AMENDMENT_DATA_SCRAMBLING.sql:1961]: Fixed

**Pull request merged**